### PR TITLE
feat: add mutation to delete existing facility

### LINKED ISSUE #844: Add delete mutation for facility to `facilitiesStore.ts`
# Problem
We have the mutation on the backend set up to delete facilities. We do not have a function to do this set up on the frontend. This will be important going forward due to so many doctors in Japan having a personal clinic. This means facilities could close.

# Success
The delete functionality for facilities is added to the `facilitiesStore.ts`
### Requirements

- [x] Add the functionality to the `facilitiesStore.ts`
- [x] Make sure is accepting the id of the facility as a parameter.

### PR DESCRIPTION
**Pull request by NabbeunNabi**
Resolves #844  
- [x] PR title follows [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specs

## 🔧 What changed
A mutation was added to delete a facility from our database. This is important because in Japan there are a lot of private clinics. They may close if a doctor retires or if the doctor decides to join another practice

## 🧪 Testing instructions
Tests will be added after the implementation of ticket #832

### CODE REVIEW

**Comment by Anissa3005** [comment]
@NabbeunNabi Maybe something we should think about is, do we want everyone that has access to the moderation panel to be able to delete a facility? Or only someone with certain authorization? Before we implement this I think we should be certain who is able to delete a facility.

**Comment by NabbeunNabi** [comment]
> @NabbeunNabi Maybe something we should think about is, do we want everyone that has access to the moderation panel to be able to delete a facility? Or only someone with certain authorization? Before we implement this I think we should be certain who is able to delete a facility.

@Anissa3005 I agree that the implementation of how this mutation is used should be discussed. This is just adding the function to the store so that when we have that answer it can be used in the correct manner. I do think that OAuth provides different levels of Auth. So if a type of authentication is sent with the mutation it should check the authorization on the backend? Or if they are admin level authentication that this is available on the frontend. I wasn't planning on using the function right away. I just want them there so when the testing ticket gets resolved we can start testing the functionality and also use it when we have the answer to your question.

**Comment by NabbeunNabi** [comment]
@ermish I also added the return type here. I looked at the backend to see what each one was returning to the frontend as a type

File: stores/facilitiesStore.ts [hunk 1]
```diff
@@ -36,23 +36,37 @@ export const useFacilitiesStore = defineStore(
...
         }
 
+        async function deleteFacility(facilityId: MutationDeleteFacilityArgs) {
+            try {
+                return await graphQLClientRequestWithRetry(
+                    gqlClient.request.bind(gqlClient),
+                    deleteExistingFacilityGqlMutation,
+                    facilityId
+                )
+            } catch (error) {
```
Comment (ermish) [inline stores/facilitiesStore.ts:59]: we should return a failure state to the calling function so they can display a nice error to the user if this fails

**Review by ermish — CHANGES_REQUESTED** [review]
🤖 Delete mode activated 🤖

File: stores/facilitiesStore.ts [hunk 2]
```diff
@@ -36,23 +36,37 @@ export const useFacilitiesStore = defineStore(
...
+                ) as DeleteResult
             } catch (error) {
                 console.error('Failed to update facility:', error)
             }
         }
 
+        async function deleteFacility(facilityId: MutationDeleteFacilityArgs) {
+            try {
+                return await graphQLClientRequestWithRetry(
+                    gqlClient.request.bind(gqlClient),
```
[the remark this answers is not in the document]
Comment (NabbeunNabi) [reply stores/facilitiesStore.ts:55]: @ermish We discussed how `graphQL` has a built in error response. So once we finish writing up the frontend pattern that is reusable, then we can add issues to apply the pattern to all of our calls

**Pull request merged**